retroshare-gui/src/gui/Posted/PostedCardView.cpp

```diff
@@ -172,90 +176,106 @@ void PostedCardView::fill()
+		qtime.setTime_t(mPost.mMeta.mPublishTs);
+		QString timestamp = qtime.toString("hh:mm dd-MMM-yyyy");
+		QString timestamp2 = misc::timeRelativeToNow(mPost.mMeta.mPublishTs);
+		ui->dateLabel->setText(timestamp2);
+		ui->dateLabel->setToolTip(timestamp);
+
+		ui->fromLabel->setId(mPost.mMeta.mAuthorId);
+
+		// Use QUrl to check/parse our URL
+		// The only combination that seems to work: load as EncodedUrl, extract toEncoded().
+		QByteArray urlarray(mPost.mLink.c_str());
+		QUrl url = QUrl::fromEncoded(urlarray.trimmed());
+		QString urlstr = "Invalid Link";
+		QString sitestr = "Invalid Link";
+
+		bool urlOkay = url.isValid();
+		if (urlOkay)
 		{
-			urlOkay = false;
-			sitestr = "Invalid Link Scheme";
+			QString scheme = url.scheme();
+			if ((scheme != "https") 
+				&& (scheme != "http")
+				&& (scheme != "ftp") 
+				&& (scheme != "retroshare")) 
+			{
+				urlOkay = false;
+				sitestr = "Invalid Link Scheme";
+			}
 		}
-	}
-    
-	if (urlOkay)
-	{
-		urlstr =  QString("<a href=\"");
-		urlstr += QString(url.toEncoded());
-		urlstr += QString("\" ><span style=\" text-decoration: underline; color:#2255AA;\"> ");
-		urlstr += messageName();
-		urlstr += QString(" </span></a>");
-
-		QString siteurl = url.toEncoded();
-		sitestr = QString("<a href=\"%1\" ><span style=\" text-decoration: underline; color:#0079d3;\"> %2 </span></a>").arg(siteurl).arg(siteurl);
 		
-		ui->titleLabel->setText(urlstr);
-	}else
-	{
-		ui->titleLabel->setText(messageName());
+		if (urlOkay)
+		{
+			urlstr =  QString("<a href=\"");
+			urlstr += QString(url.toEncoded());
+			urlstr += QString("\" ><span style=\" text-decoration: underline; color:#2255AA;\"> ");
+			urlstr += messageName();
+			urlstr += QString(" </span></a>");
+
+			QString siteurl = url.toEncoded();
+			sitestr = QString("<a href=\"%1\" ><span style=\" text-decoration: underline; color:#0079d3;\"> %2 </span></a>").arg(siteurl).arg(siteurl);
+			
+			ui->titleLabel->setText(urlstr);
+		}else
+		{
+			ui->titleLabel->setText(messageName());
 
-	}
-	
-	if (urlarray.isEmpty())
-	{
-		ui->siteLabel->hide();
-	}
+		}
+		
+		if (urlarray.isEmpty())
+		{
+			ui->siteLabel->hide();
+		}
 
-	ui->siteLabel->setText(sitestr);
-	
-	if(mPost.mImage.mData != NULL)
-	{
-		QPixmap pixmap;
-		GxsIdDetails::loadPixmapFromData(mPost.mImage.mData, mPost.mImage.mSize, pixmap,GxsIdDetails::ORIGINAL);
-		// Wiping data - as its been passed to thumbnail.
+		ui->siteLabel->setText(sitestr);
 		
-		QPixmap scaledpixmap;
-		if(pixmap.width() > 800){
-			QPixmap scaledpixmap = pixmap.scaledToWidth(800, Qt::SmoothTransformation);
-			ui->pictureLabel->setPixmap(scaledpixmap);
-		}else{
-			ui->pictureLabel->setPixmap(pixmap);
+		if(mPost.mImage.mData != NULL)
+		{
+			QPixmap pixmap;
+			GxsIdDetails::loadPixmapFromData(mPost.mImage.mData, mPost.mImage.mSize, pixmap,GxsIdDetails::ORIGINAL);
+			// Wiping data - as its been passed to thumbnail.
+			
+			QPixmap scaledpixmap;
+			if(pixmap.width() > 800){
```

Comment:
avoid the explicit size. It's better to use for instance 50*QFontMetricsF(font()).height()

Reply:
this is old code i doesnt changed it on this pr, yes it needs to be fixed with font metrics